### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by pjaselin**
- adding all cubist features (whether to use a composite model and cross-validation)
- doesn't store training data if not using a composite model
- improved text output (removed unnecessary text)
- faster post-processing (one step to get the rule summaries)
- cleaned up readme and added example output

### CODE REVIEW

**Comment by pjaselin** [comment]
sort of resolves #72

File: cubist/tests/test_cubist.py [hunk 1]
```diff
@@ -24,69 +25,103 @@ def test_model_instance(expected_output):
...
-    assert model.is_fitted_
+    check_is_fitted(model)
 
 
 def test_n_committees_fail():
     model = Cubist(n_committees=101)
     with pytest.raises(ValueError):
         model.fit(X, y)
-        assert model.is_fitted_
+        check_is_fitted(model)
```
Comment (mortenvester1) [inline cubist/tests/test_cubist.py:48]: Why does the first set of tests pass but the second doesnt?

File: cubist/src/text.h [hunk 2]
```diff
@@ -84,10 +84,10 @@
...
   "\t-h\t\tprint this message\n"
 #define T_TargetAtt "\n    Target attribute `%s'\n"
-#define TX_ReadData(c, a, f)                                                   \
-  "\nRead %d cases (%d attributes) from"                                       \
-  " %s.data\n",                                                                \
-      c, a, f
+#define TX_ReadData(c, a)                                                   \
+  "\nRead %d cases (%d attributes)"                                       \
+  "\n",                                                                \
+      c, a
```
Comment (mortenvester1) [inline cubist/src/text.h:90]: Being nitpicky here, but is there a reason for the "\" to be "right-aligned"?

File: cubist/tests/test.py [hunk 3]
```diff
@@ -1,17 +1,25 @@
 import pandas as pd
```
Comment (mortenvester1) [inline cubist/tests/test.py:1]: Is this file being used for anything? Doesn't seem like it fits with your unit tests

**Review by mortenvester1 — APPROVED** [review]
Looks good. Only minor points on the test suite really. Also is there a reason for the Makefile to be empty?

File: cubist/tests/test.py (hunk 3 above)
Comment (pjaselin) [reply cubist/tests/test.py:1]: moved to examples folder

File: cubist/src/text.h (hunk 2 above)
Comment (pjaselin) [reply cubist/src/text.h:90]: made into a one-liner

File: cubist/tests/test_cubist.py (hunk 1 above)
Comment (pjaselin) [reply cubist/tests/test_cubist.py:48]: so the R package suggests the max number of committees is 100 but the C code might not even require that, will test

Comment (mortenvester1) [reply cubist/tests/test_cubist.py:48]: Yes I noticed the checks after commenting.

**Comment by pjaselin** [comment]
> 

I actually deleted that file, it was from the R code but we don't need it here

File: cubist/cubist.py [hunk 4]
```diff
@@ -125,19 +139,78 @@ def __init__(self,
...
+        self.cv = cv
         self.random_state = random_state
         self.target_label = target_label
         self.verbose = verbose
     
     def _more_tags(self):
         return {"allow_nan": True,
                 "X_types": ["2darray", "string"]}
+    
+    def _validate_model_parameters(self):
```
Comment (mortenvester1) [inline cubist/cubist.py:154]: I like this! 

If you wanna be a bit extra, you could collect all errors and then at the end raise a CubistParameterError (or something) alerting the user of all their mistakes.

**Review by mortenvester1 — APPROVED** [review]

File: cubist/cubist.py (hunk 4 above)
Comment (pjaselin) [reply cubist/cubist.py:154]: I mean I guess I could, I should also probably make all the functions private that come from private files

Comment (pjaselin) [reply cubist/cubist.py:154]: https://stackoverflow.com/questions/29150132/collect-all-errors-and-raise-reduced-exception
maybe another time...

**Pull request merged**